core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/salt/rotatepassword/RotateSaltPasswordAction.java

```diff
@@ -0,0 +1,43 @@
+package com.sequenceiq.cloudbreak.core.flow2.cluster.salt.rotatepassword;
+
+import java.util.Map;
+import java.util.Optional;
+
+import javax.inject.Inject;
+
+import org.springframework.statemachine.StateContext;
+
+import com.sequenceiq.cloudbreak.common.event.Payload;
+import com.sequenceiq.cloudbreak.core.flow2.AbstractStackAction;
+import com.sequenceiq.cloudbreak.domain.stack.Stack;
+import com.sequenceiq.cloudbreak.reactor.api.event.cluster.RotateSaltPasswordFailureResponse;
+import com.sequenceiq.cloudbreak.reactor.api.event.cluster.RotateSaltPasswordReason;
+import com.sequenceiq.cloudbreak.service.stack.StackService;
+import com.sequenceiq.flow.core.FlowParameters;
+
+public abstract class RotateSaltPasswordAction<P extends Payload>
+        extends AbstractStackAction<RotateSaltPasswordState, RotateSaltPasswordEvent, RotateSaltPasswordContext, P> {
+
+    public static final String REASON = "reason";
+
+    @Inject
+    private StackService stackService;
+
+    protected RotateSaltPasswordAction(Class<P> payloadClass) {
+        super(payloadClass);
+    }
+
+    @Override
+    protected RotateSaltPasswordContext createFlowContext(FlowParameters flowParameters,
+            StateContext<RotateSaltPasswordState, RotateSaltPasswordEvent> stateContext, P payload) {
+        Stack stack = stackService.getByIdWithListsInTransaction(payload.getResourceId());
```

Comment:
Yes, if the Flow's actions only use the primitive fields like CRN, stackType or id, the contexts should contain the StackView only (this is the cheapest stack fetching with the primitive fields)

As I saw in most of the cases the StackView is enough because we use the CRN, name and ID only

Reply:
@Bajzathd  so please fix this as recommended by Topi